numpy/core/shape_base.py

```diff
@@ -275,3 +278,67 @@ def hstack(tup):
         return _nx.concatenate(arrs, 0)
     else:
         return _nx.concatenate(arrs, 1)
+
+def stack(arrays, axis=0):
+    """
+    Join a sequence of arrays along a new axis.
```

Comment:
Need more explanation here. For instance in the examples, it isn't clear why `np.stack((a, b))` is not the same as `np.stack((a, b), axis-1)`.